lib/FrameManager.js

```diff
@@ -566,8 +566,15 @@ class Frame {
    * @return {!Promise}
    */
   waitFor(selectorOrFunctionOrTimeout, options = {}, ...args) {
-    if (helper.isString(selectorOrFunctionOrTimeout))
-      return this.waitForSelector(/** @type {string} */(selectorOrFunctionOrTimeout), options);
+    const xpathPattern = /^\//;
+
+    if (helper.isString(selectorOrFunctionOrTimeout)) {
+      const string = /** @type {string} */ (selectorOrFunctionOrTimeout);
+      if (string.match(xpathPattern))
+        return this.waitForXPath(string, options);
+      else
```

Comment:
style: drop `else`. (We follow [blink's codestyle](https://www.chromium.org/blink/coding-style))

Reply:
done.